### PR TITLE
Hmx linux 6.1.22 kirkstone

### PR DESCRIPTION
**Pull request by mattiasbus**
Use 6.1 kernel for HMX.
known issue: tcan4x5x driver does not go into low power mode (sleep). Code needs to be added to do this which includes restoring chip registers in resume function.

### CODE REVIEW

File: dynamic-layers/recipes-kernel/linux/linux-variscite/0001-add-hmx-device-tree.patch [hunk 1]
```diff
@@ -55,28 +55,35 @@ v13. Alias ethernet2 as eth0_obd and ethernet3 as eth1_rj45 to only need 2
 
 v14. Add RGB led driver (only for i2c 0x30),
 the leds will be indicator:red/green/blue
+
+v15. Use 8 frames tx fifo for tcan4x5x.
+
+v16. Add wakeup-source property to tcan4x5x nodes. This is used by new
+     driver patch to allow wake-up when interface in up when going into
+     suspend.
```
Comment (albsod) [inline dynamic-layers/recipes-kernel/linux/linux-variscite/0001-add-hmx-device-tree.patch:63]: in up -> is up?

File: dynamic-layers/recipes-kernel/linux/linux-variscite/0030-gpio-pca953x-driver-minimize-error-print-out.patch [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
 +++ b/drivers/gpio/gpio-pca953x.c
-@@ -390,7 +390,8 @@ static u8 pca953x_recalc_addr(struct pca953x_chip *chip, int reg, int off)
- 
- 	return regaddr;
+@@ -482,6 +482,8 @@ static u8 pcal6534_recalc_addr(struct pca953x_chip *chip, int reg, int off)
+ 	return pinctrl + addr + (off / BANK_SZ);
  }
--
+ 
 +// HACK: We dont want the chip to send to much error message when the HMX unit goes to super cap mode(shutdown with no power to the board)
```
Comment (albsod) [inline dynamic-layers/recipes-kernel/linux/linux-variscite/0030-gpio-pca953x-driver-minimize-error-print-out.patch:19]:
```suggestion
+// HACK: We don't want the chip to send many error messages when the HMX unit goes to super cap mode (shutdown with no power to the board)
```

File: dynamic-layers/recipes-kernel/linux/linux-variscite/0036-m_can-dont-enable-transceiver-when-probing.patch [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+This has same the purpose as
+0026-HMX-m_can-do-not-init-registers-until-bus-on.patch but for 6.1.
+
+From:
+https://www.spinics.net/lists/linux-can/msg15155.html
+
+The m_can driver sets and clears the CCCR.INIT bit during probe (both
+when testing the NON-ISO bit, and when configuring the chip). After
+clearing the CCCR.INIT bit, the transceiver enters normal mode, where it
+affects the CAN bus (i.e. ACKs frames). This can cause trouble when the
```
Comment (albsod) [inline dynamic-layers/recipes-kernel/linux/linux-variscite/0036-m_can-dont-enable-transceiver-when-probing.patch:18]: i.e. -> e.g.

?

**Comment by albsod** [comment]
LGTM

File: dynamic-layers/recipes-kernel/linux/linux-variscite_%.bbappend [hunk 4]
```diff
@@ -8,22 +8,36 @@ file://0002-Set-I2C1-for-imx8mp-var-dart.dtsi-to-lower-clock-fre.patch \
...
 file://0022-gpio-keys-make-disabled-keys-not-wake-system.patch  \
+file://0030-gpio-pca953x-driver-minimize-error-print-out.patch \
+file://0001-Add-Kinetic-KTD2026-2027-LED-driver.patch \
+file://0034-tcan4x5x-Backport-linuxcan-next-6.6-driver.patch \
+file://0035-tcan4x5x-remove-msg-lost-in-rxf0-netdev_err.patch \
+file://0036-m_can-dont-enable-transceiver-when-probing.patch \
+file://0037-HMX-tcan4x5x-enable-wakeup-when-interface-is-up.patch \
+"
+
+# TODO fix rest of CAN
```
Comment (RikardO-HM) [inline dynamic-layers/recipes-kernel/linux/linux-variscite_%.bbappend:19]: I assume you have add the can so remove UNUSED_PATCHES

**Pull request merged**